examples/rcbd_two_var.R

```diff
@@ -0,0 +1,179 @@
+shapiro.test(residuals(model))
+#------------------------------------------------------------------------------#
+sep(79)
+
+# Run Levene's Test for a one-way ANOVA of each of the main factors.
+# TODO : clone is significant in Levene's test so transformation is necessary.
+cat("Levene's Test\n")
+sep(79)
+#------------------------------------------------------------------------------#
+leveneTest(yield ~ clone, data=my.data)
+leveneTest(yield ~ nitrogen, data=my.data)
+#------------------------------------------------------------------------------#
+sep(79)
+
+# Perform a Tukey 1-df Test for Non-additivity
+# TODO : It may be necessary to use as.factor() on the factor variables if we
+# have to use lm() here, since squared.preds is continuous.
+cat("Tukey 1-df Test for Non-additivity\n")
+sep(79)
+#------------------------------------------------------------------------------#
+my.data$squared.preds <- predict(model)^2
+one.df.model <- lm(log10.yield ~ clone + nitrogen + clone:nitrogen + block +
+                   squared.preds, my.data)
+anova(one.df.model)
+#------------------------------------------------------------------------------#
+sep(79)
+
+# Print the ANOVA table of the fit. The user will have to note the significant
+# factors. In this case all factors, including interaction nitrogen:clone, are
+# significant.
+cat('ANOVA Table\n')
+sep(79)
+#------------------------------------------------------------------------------#
+anova(model)
+#------------------------------------------------------------------------------#
+sep(79)
+
+# Plot the mean yield with respect to each clone for each N level and vice
+# versa.
+dev.new()
+#------------------------------------------------------------------------------#
+intxplot(log10.yield ~ clone, groups = nitrogen, data = my.data, se = TRUE,
+         ylim = range(my.data$log10.yield), offset.scale = 500)
+#------------------------------------------------------------------------------#
+invisible(dev.copy(png, 'rcbd-two-var-clone-int-plot.png'))
+invisible(dev.off())
+dev.new()
+#------------------------------------------------------------------------------#
+intxplot(log10.yield ~ nitrogen, groups = clone, data = my.data, se = TRUE,
+         ylim = range(my.data$log10.yield), offset.scale = 500)
+#------------------------------------------------------------------------------#
+invisible(dev.copy(png, 'rcbd-two-var-nitrogen-int-plot.png'))
+invisible(dev.off())
+
+# The ANOVA table shows that each variable and the interaction are significant,
+# so we then see which levels of both clone and nitrogen are significant with
+# respect to each other using least significant difference.
+cat('Least Significant Difference\n')
+sep(79)
+#------------------------------------------------------------------------------#
+lsd.results <- LSD.test(model, c('clone', 'nitrogen'), console = TRUE)
+#------------------------------------------------------------------------------#
```

Comment:
Since talking with Lauren, we decided that we don't need to back-transform the LS means (lsd.results$means$yield or lsd.results$groups$means). However, what we want here is for the user to view 2 outputs: First, a table for the lsd.results of the transformed dataset (done in line 148), and second, a partial amended table of the lsd.results of the original dataset. For this second table based on the original dataset: i.e., aov(yield ~ block + clone + nitrogen + clone:nitrogen, data=my.data), we want the user to see the lsd.results$means table with the corresponding LSD letters (lsd.results$groups$M) from the transformed dataset copied to it, by matching treatments. The second dataframe would be used for generating the post-hoc plot(s). 

This is where I need your help. I've been working on creating the second table, but since there isn't a column within lsd.results$means specifying the treatment names, I don't know how join the column lsd.results$groups$M from transformed dataframe to the original dataframe lsd.results$means using 'by' statement. 

Whatever we create from this, will be fed into the ggplot, and I want to make sure that we don't show or use the lsd letters from the original dataset.

Not sure how to do this.

This is the workflow I have so far:

# When there is a transformed variable, as in this example, do second lsd.test on original dataset to obtain dataframe       
# lsd.results.original, but replace lsd.results.original$groups$M with lsd.results$groups$M from transformed dataset. 

# Do lsd test on original data
model.original <- aov(yield ~ block + clone + nitrogen + clone:nitrogen,
             data=my.data)
get.orig.LSmeans <- LSD.test(model.original, c('clone', 'nitrogen')) #Don't show output yet
# Replace get.orig.LSmeans$groups$M with lsd.test$groups$M
??
# Show single table with LS mean, std, r, LCL, UCL, min, max, and lsd.test$groups$M. Display with heading: 
 # 'Lease square means of original dataset. Note that LSD test was performed on log-transformed dataset.' Followed by 'ATTENTION: When reporting LSD groups in the original units shown here, remember to specify that the LSD test was performed on log-transformed data.'